iDiff/differs/pipDiff.go

```diff
@@ -14,71 +14,77 @@ type PipDiffer struct {
 
 // PipDiff compares pip-installed Python packages between layers of two different images.
 func (d PipDiffer) Diff(image1, image2 utils.Image) (utils.DiffResult, error) {
-	diff, err := singleVersionDiff(image1, image2, d)
+	diff, err := multiVersionDiff(image1, image2, d)
 	return diff, err
 }
 
-func getPythonVersion(pathToLayer string) (string, bool) {
-	libPath := filepath.Join(pathToLayer, "/layer/usr/local/lib")
+func getPythonVersion(pathToLayer string) ([]string, error) {
+	matches := []string{}
+	libPath := filepath.Join(pathToLayer, "usr/local/lib")
 	libContents, err := ioutil.ReadDir(libPath)
 	if err != nil {
-		return "", false
+		return matches, err
 	}
 
 	for _, file := range libContents {
 		pattern := regexp.MustCompile("^python[0-9]+\\.[0-9]+$")
 		match := pattern.FindString(file.Name())
 		if match != "" {
-			return match, true
+			matches = append(matches, match)
 		}
 	}
-	return "", false
+	return matches, nil
 }
 
-func (d PipDiffer) getPackages(path string) (map[string]utils.PackageInfo, error) {
-	packages := make(map[string]utils.PackageInfo)
+func (d PipDiffer) getPackages(path string) (map[string]map[string]utils.PackageInfo, error) {
+	packages := make(map[string]map[string]utils.PackageInfo)
 
-	// TODO: Eventually, this would make use of the shallow JSON and be diffed
-	// with that of another image to get only the layers that have changed.
-	layers := utils.GetImageLayers(path)
-	for _, layer := range layers {
-		pathToLayer := filepath.Join(path, layer)
-		pythonVersion, exists := getPythonVersion(pathToLayer)
-		if !exists {
-			// layer doesn't have a Python folder installed
-			continue
-		}
-		packagesPath := filepath.Join(pathToLayer, "layer/usr/local/lib", pythonVersion, "site-packages")
+	pythonVersions, err := getPythonVersion(path)
+	if err != nil {
+		// layer doesn't have a Python version installed
+		return packages, nil
+	}
+	for _, pyVersion := range pythonVersions {
+		packagesPath := filepath.Join(path, "usr/local/lib", pyVersion, "site-packages")
 		contents, err := ioutil.ReadDir(packagesPath)
 		if err != nil {
 			// layer's Python folder doesn't have a site-packages folder
-			continue
+			return packages, nil
```

Comment:
by default, pip just puts all packages in site-packages. but you can configure it to install them anywhere, for example by passing `--target=/some/random/directory` to `pip install` and then adding that directory to your `PYTHONPATH`. it might be the case that we want to check all directories on the `PYTHONPATH` when we do the python diff.

check out [this StackOverflow post](https://stackoverflow.com/questions/2915471/install-a-python-package-into-a-different-directory-using-pip) for more info on this